python/caffe/test/test_python_layer_with_param_str.py

```diff
@@ -0,0 +1,85 @@
+import unittest
+import tempfile
+import os
+import six
+
+import caffe
+
+class ElemwiseScalarLayer(caffe.Layer):
+    """A layer that just multiplies by ten"""
+
+    def setup(self, bottom, top):
+        self.layer_params_ = eval(self.param_str_)
+        self.value_ = self.layer_params_['value']
+        if self.layer_params_['op'].lower() == 'add':
+            self._forward = self._forward_add
+            self._backward = self._backward_add
+        elif self.layer_params_['op'].lower() == 'mul':
+            self._forward = self._forward_mul
+            self._backward = self._backward_mul
+        else:
+            raise ValueError("Unknown operation type: '%s'"
+                % self.layer_params_['op'].lower())
+    def _forward_add(self, bottom, top):
+        top[0].data[...] = bottom[0].data + self.value_
+    def _backward_add(self, bottom, propagate_down, top):
+        bottom[0].diff[...] = top[0].diff
+    def _forward_mul(self, bottom, top):
+        top[0].data[...] = bottom[0].data * self.value_
+    def _backward_mul(self, bottom, propagate_down, top):
+        bottom[0].diff[...] = top[0].diff * self.value_
+    def reshape(self, bottom, top):
+        top[0].reshape(bottom[0].num, bottom[0].channels, bottom[0].height,
+                bottom[0].width)
+    def forward(self, bottom, top):
+        self._forward(bottom, top)
+    def backward(self, top, propagate_down, bottom):
+        self._backward(bottom, propagate_down, top)
+
+
+def python_net_file():
+    f = tempfile.NamedTemporaryFile(delete=False)
```

Comment:
```NamedTemporaryFile``` is created with ```mode='w+b'``` by default, which in python3 is a buffer, not a string. To fix this just call ```NamedTemporaryFile(mode='w+', delete=False)```.